src/libostree/ostree-sign-x509.c

```diff
@@ -0,0 +1,651 @@
+{
+  g_assert (OSTREE_IS_SIGN (self));
+
+  OstreeSignX509 *sign = _ostree_sign_x509_get_instance_private (OSTREE_SIGN_X509 (self));
+
+  g_autofree guint8 *key_owned = NULL;
+  const guint8 *key = NULL;
+  gsize n_elements = 0;
+
+  if (g_variant_is_of_type (revoked_key, G_VARIANT_TYPE_STRING))
+    {
+      const gchar *rk_ascii = g_variant_get_string (revoked_key, NULL);
+      key = key_owned = g_base64_decode (rk_ascii, &n_elements);
+    }
+  else if (g_variant_is_of_type (revoked_key, G_VARIANT_TYPE_BYTESTRING))
+    {
+      key = g_variant_get_fixed_array (revoked_key, &n_elements, sizeof (guchar));
+    }
+  else
+    {
+      return glnx_throw (error, "Unknown x509 revoked key type");
+    }
+
+  g_autofree char *hex = g_malloc0 (n_elements * 2 + 1);
+  ot_bin2hex (hex, key, n_elements);
+  g_debug ("Read x509 revoked key = %s", hex);
+
+  g_autoptr (GBytes) key_bytes = g_bytes_new_static (key, n_elements);
+  if (g_list_find_custom (sign->revoked_keys, key, g_bytes_compare) == NULL)
+    {
+      GBytes *new_key_bytes = g_bytes_new (key, n_elements);
+      sign->revoked_keys = g_list_prepend (sign->revoked_keys, new_key_bytes);
+    }
+
+  return TRUE;
+}
+
+static gboolean
+_load_pk_from_stream (OstreeSign *self, GInputStream *key_stream_in, gboolean trusted,
+                      GError **error)
+{
+  if (key_stream_in == NULL)
+    return glnx_throw (error, "x509: unable to read from NULL key-data input stream");
+
+  gboolean ret = FALSE;
+
+  g_autoptr (OstreeBlobReader) blob_reader = ostree_sign_read_pk (self, key_stream_in);
+  g_assert (blob_reader);
+
+  /* Use simple file format with just a list of base64 public keys per line */
+  while (TRUE)
+    {
+      g_autoptr (GVariant) pk = NULL;
+      gboolean added = FALSE;
+      g_autoptr (GError) local_error = NULL;
+      g_autoptr (GBytes) blob = ostree_blob_reader_read_blob (blob_reader, NULL, &local_error);
+
+      if (local_error != NULL)
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
+        }
+
+      if (blob == NULL)
+        return ret;
+
+      /* Read the key itself */
+      pk = g_variant_new_fixed_array (G_VARIANT_TYPE ("y"), g_bytes_get_data (blob, NULL),
+                                      g_bytes_get_size (blob), sizeof (guchar));
+
+      if (trusted)
+        added = ostree_sign_x509_add_pk (self, pk, error);
+      else
+        added = _x509_add_revoked (self, pk, error);
+
+      g_autofree gchar *pk_printable = g_variant_print (pk, FALSE);
+      g_debug ("%s %s key: %s", added ? "Added" : "Invalid", trusted ? "public" : "revoked",
+               pk_printable);
+
+      /* Mark what we load at least one key */
+      if (added)
+        ret = TRUE;
+    }
+
+  return ret;
+}
+
+static gboolean
+_load_pk_from_file (OstreeSign *self, const gchar *filename, gboolean trusted, GError **error)
+{
+  g_debug ("Processing file '%s'", filename);
+
+  g_autoptr (GFile) keyfile = NULL;
```

Comment:
Here (and a lot of other places) - our new preferred coding style is declare-and-initialize as an atomic unit. In this specific function we clearly can do so.

Not a blocker at all, just a general note.

Reply:
Fixed.